### ISSUE TITLE
Co-simulation does not handle signal value type properly 

### ISSUE THREAD

**Issue by hashhsah** [jandecaluwe/myhdl#5]
Symptom
-------------

If a signal has boolean initial value, and during co-simulation, external simulator sets its value to 'X', the simulation fails with 
```
ERROR: testCoSimStatic (__main__.TestTop)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testbench.py", line 185, in testCoSimStatic
    sim.run()
  File "/home/hash/src/HISEE_Test/myhdl/venv/local/lib/python2.7/site-packages/myhdl/_Simulation.py", line 126, in run
    _extend(s._update())
  File "/home/hash/src/HISEE_Test/myhdl/venv/local/lib/python2.7/site-packages/myhdl/_Signal.py", line 191, in _update
    self._val._val = next._val
AttributeError: 'int' object has no attribute '_val'

----------------------------------------------------------------------
```
  
Root cause:
----------------

In `Cosimulation._get()`,  _Cosimulation.py:134, we have
```python
            s, v = self._toSigDict[e[i]], e[i+1]
            try:
                next = int(v, 16)
                if s._nrbits and s._min is not None and s._min < 0:
                    if next >= (1 << (s._nrbits-1)):
                        next |= (-1 << s._nrbits)
            except ValueError:
                next = intbv(0)
            s.next = next
```
If external simulator returns anything that does not converts to integer, `intbv(0)` is assigned to `s.next`.

In  `_Signal._update()`, we have
```python
            if next is None:
                self._val = None
            elif isinstance(val, intbv):
                self._val._val = next._val
            elif isinstance(val, (int, long, EnumItemType)):
                self._val = next
            else:
                self._val = deepcopy(next)
```
Since the signal has boolean initial value, the `else` clause is picked, and the type of `self._val` changed from `bool` to `intbv`.

Some time later, when the external simulator returns a valid integer value,  `Cosimulation._get()` assigns `next` the integer. Now we encounter the problem in `_Signal._update()`: we enter the `intbv` clause, but `next` is an integer instead of `intbv`.
 
Possible fix:
----------------

I added the following clause before the `intbv` clause in `_Signal._update()`.
```python
            elif isinstance(val, bool):
                self._val = bool(next)
            elif isinstance(val, intbv):
                self._val._val = next._val
```
It worked for me. However, we can also fix it in `Cosimulation._get()`. 

The type handling in Signal assignment is a bit confusing, I'm not sure if this fix is correct.

**Comment by hashhsah** [comment]
I wrote a testcase to demonstrate this issue (cogenda/myhdl@b59c8d7fb7f7ad17eff585a62755dd44c3fe23db) , and an incomplete fix ( cogenda/myhdl@191d5a55c8a47b219ccd5504a3f12b97eee9be1c ).

**Comment by hashhsah** [comment]
This bug is related to issue #12.  @524a2ae also fixed this bug. I'm closing this.
